src/rpc_server.rs

```diff
@@ -1696,6 +1735,14 @@ pub trait RPC {
     /// ```
     async fn restart_miner(token: rpc_auth::Token) -> RpcResult<()>;
 
+    /// Provide a PoW-solution to the current block proposal.
+    ///
+    /// Returns true iff the provided solution satisfies the proof-of-work
+    /// threshold of the next block. False otherwise. False can be returned if
+    /// the solution comes in too late in which case either a new block
+    /// proposal is stored by the client, or no block proposal is known at all.
+    async fn pow_solution(token: rpc_auth::Token, nonce: Digest) -> RpcResult<bool>;
```

Comment:
With this function name, I'd expect the return type to be the pow_solution.

Maybe one of the following works better?
```suggestion
    async fn is_pow_solution(token: rpc_auth::Token, nonce: Digest) -> RpcResult<bool>;
```
```suggestion
    async fn pow_solution(token: rpc_auth::Token, nonce: Digest) -> RpcResult<Result<(), ()>>;
```
```suggestion
    async fn solve_pow_puzzle(token: rpc_auth::Token, nonce: Digest) -> RpcResult<Result<(), ()>>;
```

Reply:
I'm happy with `provide_pow_solution`